test/components/popup.spec.tsx

```diff
@@ -29,120 +35,125 @@ describe('<Popup />', () => {
+                <span>Popup Body</span>
             </Popup>
         );
 
+        const popup = new PopupTestDriver(result as Popup);
+
         return waitFor(() => {
-            expect(bodySelect(portalId)).to.be.present();
-            expect(bodySelect(portalId, 'SPAN')).to.be.present();
+            expect(popup.root).to.be.present();
+            expect(popup.content[0]).to.be.present();
         });
     });
 
     it('does not render the popup if there is no anchor', async () => {
-        clientRenderer.render(
-            <Popup anchor={null} open={true}>
-                <span data-automation-id="SPAN">Popup Body</span>
+        const {result} = clientRenderer.render(
+            <Popup anchor={null} open>
+                <span>Popup Body</span>
             </Popup>
         );
         await sleep(100);
-        await waitFor(() => expect(bodySelect(portalId)).to.be.absent());
-    });
 
-    it('does not reder the popup if the open prop is false', async () => {
-        clientRenderer.render(
-            <Popup anchor={anchor} open={false}>
-                <span data-automation-id="SPAN">Popup Body</span>
-            </Popup>
-        );
-        await sleep(100);
-        await waitFor(() => expect(bodySelect(portalId)).to.be.absent());
+        const popup = new PopupTestDriver(result as Popup);
+        await waitFor(() => expect(popup.root).to.equal(null));
     });
 
-    it('renders the popup using a point as anchor', async () => {
-        const point: Point = {x: 100, y: 100};
-        clientRenderer.render(
-            <Popup anchor={point} open>
+    it('does not render the popup if the open prop is false', async () => {
+        const {result} = clientRenderer.render(
+            <Popup anchor={anchor}>
                 <span data-automation-id="SPAN">Popup Body</span>
             </Popup>
         );
+        await sleep(100);
 
-        await waitFor(() => expect(bodySelect(portalId)).to.be.present());
+        const popup = new PopupTestDriver(result as Popup);
+        await waitFor(() => expect(popup.root).to.equal(null));
     });
 
     it('removes the component when unmounting', async () => {
-        clientRenderer.render(
+        const {result} = clientRenderer.render(
             <Popup
                 anchor={anchor}
-                open={true}
+                open
             >
                 <span data-automation-id="SPAN">Popup Body</span>
             </Popup>);
 
-        await waitFor(() => {expect(bodySelect(portalId)).to.be.present(); });
-        ReactDOM.unmountComponentAtNode(bodySelect(portalId)!.parentElement!);
-        return waitFor(() => {expect(bodySelect(portalId)).to.not.exist; });
+        const popup = new PopupTestDriver(result as Popup);
+
+        await waitFor(() => {expect(popup.root).to.be.present(); });
+        ReactDOM.unmountComponentAtNode(popup.root!.parentElement!);
+        return waitFor(() => {expect(popup.root).to.not.exist; });
```

Comment:
please decide on a single format to check existence of `popup.root` . for instance for non-existence I see both `be.absent()` and `not.exist` and `equal(null)` ... same thing for present()

Reply:
all are using present/absent now